src/management-system-v2/lib/sharing/process-sharing.ts

```diff
@@ -0,0 +1,75 @@
+'use server';
+
+import jwt from 'jsonwebtoken';
+import { updateProcessShareInfo } from '../data/processes';
+import { Environment } from '../data/environment-schema';
+import { getEnvironmentById } from '../data/legacy/iam/environments';
+import { getUserOrganizationEnviroments } from '../data/legacy/iam/memberships';
+
+export interface TokenPayload {
+  processId: string | string[];
+  embeddedMode?: boolean;
+  timestamp?: number;
+}
+
+export interface ProcessGuestAccessRights {
+  shared?: boolean;
+  sharedAs?: 'public' | 'protected';
+  shareTimestamp?: number;
+  allowIframeTimestamp?: number;
+}
+
+export async function updateProcessGuestAccessRights(
+  processId: string | string[],
+  newMeta: ProcessGuestAccessRights,
+  spaceId: string,
+) {
+  await updateProcessShareInfo(
+    processId as string,
+    newMeta.sharedAs,
+    newMeta.shareTimestamp,
+    newMeta.allowIframeTimestamp,
+    spaceId,
+  );
+}
+
+export async function generateProcessShareToken(
+  payload: TokenPayload,
+  spaceId: string,
+  oldTimestamp?: number,
+) {
+  const secretKey = process.env.JWT_SHARE_SECRET;
+
+  let timestamp = 0;
+
+  if (oldTimestamp) {
```

Comment:
Could potentially be shortened to `const timestamp = oldTimestamp ?? Date.now();`